plugins/lookup/env_freeipa_domain.py

```diff
@@ -54,37 +55,42 @@
 
 RETURN = '''
   _list:
-    description: List of Freeipa domains for selected Environments
+    description: List of FreeIPA domains for selected Environments
     type: list
     elements: list
 '''
+
 from ansible.errors import AnsibleError
 from ansible.plugins.lookup import LookupBase
 from ansible.module_utils.common.text.converters import to_text, to_native
 from ansible.utils.display import Display
+
 from cdpy.cdpy import Cdpy
 from cdpy.common import CdpError
 
 
 display = Display()
+
 class LookupModule(LookupBase):
     def run(self, terms, variables=None, **kwargs):
         self.set_options(var_options=variables, direct=kwargs)    
+
         try: 
             results = []
             for term in LookupBase._flatten(terms):
                 environment = Cdpy().environments.describe_environment(term)
                 freeipa_client_domain = environment['freeipa']['domain']
-                results = [{'domain': freeipa_client_domain}]
-
-            if self.get_option('detailed'):
-                for term in LookupBase._flatten(terms):
-                    environment = Cdpy().environments.describe_environment(term)   
+                
+                if self.get_option('detailed'):
                     server_ips = environment['freeipa']['serverIP']
-                    results = [{'domain': freeipa_client_domain,'server_ips' : server_ips}]     
+                    results = [{'domain': freeipa_client_domain,'server_ips' : server_ips}]    
+                else:
+                    results = [{'domain': freeipa_client_domain}]
```

Comment:
Have `detailed=False` return a string, not a dict with a single key. Just test with `wantlist=True` as well as `wantlist=False` to confirm this operation.